test/services/gatherers/vote_account_details_service_test.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+class VoteAccountDetailsServiceTest < ActiveSupport::TestCase
+  setup do
+    @network = "testnet"
+    @validator_identity = "validatoridentity"
+    @authorized_withdrawer = "authorizedwithdrawer"
+    @testnet_url = "https://api.testnet.solana.com"
+
+    @json_response = {
+      validatorIdentity: @validator_identity,
+      authorizedWithdrawer: @authorized_withdrawer
+    }.to_json
+
+    @val = create(:validator, network: @network)
+    @score = create(:validator_score_v1, network: @network, validator: @val)
+  end
+
+  teardown do
+    VoteAccount.delete_all
+  end
+
+  test "VoteAccountDetailsService updates identity and withdrawer" do
+    va1 = create(
+      :vote_account,
+      validator: @val,
+      network: @network,
+      account: "acc1"
+    )
+    SolanaCliService.stub(:request, @json_response, ["vote-account", @testnet_url]) do
+      Gatherers::VoteAccountDetailsService.new(network: @network, config_urls: [@testnet_url]).call
+
+      assert_equal va1.reload.validator_identity, @validator_identity
+      assert_equal va1.reload.authorized_withdrawer, @authorized_withdrawer
+      assert_equal 0, @score.reload.authorized_withdrawer_score
+    end
+  end
+
+  test "VoteAccountDetailsService sets correct score" do
+    va1 = create(
+      :vote_account,
+      validator: @val,
+      network: @network,
+      account: "acc1",
```

Comment:
I think this could be initialized in setup so you can use it in all tests, and two attributes below you can update here in test.